ArmRobot/Assets/Scripts/RandomizerPositionObject.cs

```diff
@@ -0,0 +1,165 @@
+        Vector3 tableCenter = tableBounds.center;
+        float x = tableCenter.x + tableTopPoint.x;
+        float z = tableCenter.z + tableTopPoint.y;
+        transform.position = new Vector3(x, targetY, z);
+
+        // random rotation
+        Vector3 randomRotation = new Vector3(
+            transform.rotation.eulerAngles.x,
+            Random.value * 360.0f,
+            transform.rotation.eulerAngles.z);
+        transform.rotation = Quaternion.Euler(randomRotation);
+    }
+
+
+    // HELPERS
+
+    Vector2 RandomReachablePointOnTable(List<GameObject> listOfAlreadyMovedObjects)
+    {
+        while (true)
+        {
+            Vector2 randomOffset = RandomPoint(robotMinReach, robotMaxReach, listOfAlreadyMovedObjects);
+            bool onTable = PointOnTable(randomOffset);
+            if (onTable)
+            {
+                return randomOffset;
+            }
+        }
+    }
+
+    bool PointOnTable(Vector2 point)
+    {
+        /*  point: The 2D point on table top, relative to center of table top.
+         *  Determines if this point would be on the table or not.      
+         */
+        Vector3 tableExtents = tableBounds.extents;
+        float targetRadius = 0f;
+        if (tag == "Sphere"){
+                targetRadius = GetComponent<SphereCollider>().radius * GetComponent<Transform>().localScale[0]; // for the sphere 
+            }
+            // otherwise it is a cylinder 
+            else{
+                targetRadius = GetComponent<CapsuleCollider>().radius * GetComponent<Transform>().localScale[0]; 
+            }
+
+        float safeXDistance = tableExtents.x - targetRadius;
+        float safeZDistance = tableExtents.z - targetRadius;
+        float xDistance = Mathf.Abs(point.x);
+        float yDistance = Mathf.Abs(point.y);
+        bool onTable = (xDistance < safeXDistance) && (yDistance < safeZDistance);
+        return onTable;
+    }
+
+    public Vector2 RandomPoint(float minRadius, float maxRadius, List<GameObject> listOfAlreadyMovedObjects)
+    {
+        /*  Picks a 2D point randomly at uniform. Must be between minRadius and
+         *  maxRadius from center and not at the same place than the cube. 
+         *  Point given relative to center.      
+         */
+        //for (var j = 0; j < 100; j++)
+        while (true)
+        {
+            // pick a random point between the square of edge 2*minRadius and the one with edge 2*maxRadius to be sure that the point is reachable by the robot 
+            float randomX = 0.25f;
+            float randomZ = 0.25f;
+            float randomSignX = Random.value;
+            float criticalValueForSign = 0.5f;
+            if (randomSignX > criticalValueForSign){
+                randomX = minRadius + (maxRadius - minRadius) * Random.value;
+            }
+            else {
+                randomX = -maxRadius + (-minRadius - (-maxRadius)) * Random.value;
+            }
```

Comment:
The logic in this if / else section can be simplified. You could use `int random_sign = 2 *Random.Range(0,2) - 1` to randomly get a 1 or -1, then you don't need the if/else at all. Also, be sure to simplify the math before you write it down (e.g., use a plus rather than minus a negative - they may be equal, but the simpler form is easier to read).

Reply:
Same for the way you're computing `randomZ` down below.

Reply:
Thanks for the tips for the random sign! Also for simplifying the maths, I have voluntarily did that so that people who read it will better understand the logic behind but it seems not ahah, I will change it!

Reply:
So, the purpose of my comment was to help you replace the if/else block, which still remains. If had just one line: `float randomX = randomSignX * maxRadius + gameObjectRadius + (maxRadius - minRadius - 2 * gameObjectRadius) * Random.value;`, then you don't need an if/else here at all.

Reply:
We can't avoid the if else statement because the formula is not symmetric by sign...

Reply:
Got it. :)